CMakeLists.txt

```diff
@@ -31,24 +31,28 @@ option(TEST_COVERAGE "Enable test runtime coverage logging" OFF)
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
-# Ingest the git tag as project version name
-find_package(Git)
-execute_process(
-    COMMAND ${GIT_EXECUTABLE} describe --always --tags --dirty
-    RESULT_VARIABLE GIT_TAG_EXIT
-    OUTPUT_VARIABLE GIT_TAG_REV
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-if(NOT GIT_TAG_EXIT EQUAL 0)
-    set(GIT_TAG_REV "v0.0.0")
-    message(WARNING "No git tag found, marking as ${GIT_TAG_REV}")
+if(NOT PROJECT_VERSION)
+  # Ingest the git tag as project version name
+  find_package(Git)
+  execute_process(
+      COMMAND ${GIT_EXECUTABLE} describe --always --tags --dirty
+      RESULT_VARIABLE GIT_TAG_EXIT
+      OUTPUT_VARIABLE GIT_TAG_REV
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(NOT GIT_TAG_EXIT EQUAL 0)
+      set(GIT_TAG_REV "v0.0.0")
+      message(WARNING "No git tag found, marking as ${GIT_TAG_REV}")
+  endif()
+  # Make version compatible with CMake needs
+  STRING(REGEX REPLACE [[^v([0-9]+\.[0-9]+\.[0-9]+).*]] [[\1]] GIT_TAG_VERS "${GIT_TAG_REV}")
+  message(STATUS "Using git label ${GIT_TAG_REV} to version ${GIT_TAG_VERS}")
+
+  set(PROJECT_VERSION ${GIT_TAG_VERS})
 endif()
```

Comment:
For simple conditions please add the condition to `endif()` for readability.

Reply:
Added a header comment explaining `PROJECT_VERSION` and updated the corresponding `endif` called out above.